Sources/MongoSwift/MongoClient.swift

```diff
@@ -221,7 +229,11 @@ public class MongoClient {
         initializeMongoc()
 
         let connString = try ConnectionString(connectionString, options: options)
-        self.connectionPool = try ConnectionPool(from: connString, options: options?.tlsOptions)
+        self.connectionPool = try ConnectionPool(
+            from: connString,
+            poolOptions: options?.connectionPoolOptions,
```

Comment:
Related to above: if you move everything directly into `ClientOptions` you can more easily pass that single unit around, rather than needing to split up parts like this.